### PR TITLE
Don't allow new items to use legacy int to_hit

### LINKED ISSUE #74563: Begin enforcing the new ``to-hit`` syntax
### Is your feature request related to a problem? Please describe.

For a longer while now we've had the fancy new ``to-hit`` syntax all ready to go. It's awesome and 9/10 times easy to use. Except that almost no newly added items seem to use it.

### Solution you would like.

Do something similar to the known bad density list for old ``to-hit`` syntax - list all items using old syntax there, and throw an error if someone tries to add the old syntax to more items for any reason. Important to have this list ignore items without ``to-hit`` defined at all, so as to not force every single item to have the field defined - it's pointless for at least a solid half of our itempool that isn't meant to be used as weapons.

### Describe alternatives you have considered.

Enforce the need to have a ``to-hit`` defined for any items with melee damage defined as well. I think it's not a bad idea, since if it's got melee damage defined it's expected to be usable as a weapon, and as such should have these values defined.

### Additional context

@Procyonae you asked me to fill this as an issue

Enforcing this syntax would allow us to make wider use of the variables it defines, such as using the contact surface for weakpoints, length for disarming, or balance for potentially losing grip on your weapon. These are just examples, but there's been a nonzero amount of conversations on the devcord where using those variables came up as possibly being useful.

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change
Closes #74563

#### Describe the solution
Adds a test with blacklist that flags any new items being added using the legacy method.
Changes the default to_hit to -2 as intended and remove all explicit -2 to_hit
Adds to_hit to various "weapons" lacking it so that the default change doesn't unintentionally nerf them
Adds initial blacklists for vanilla and all repo mods that need updating, deals with mods with very few issues now rather than adding them to blacklists

#### Describe alternatives you've considered
Keeping legacy stuff around forever

#### Testing
Ran the test on every in repo mod separately
Ran a seperate (not included) test to determine "weapons" lacking to_hit to determine where it needed manually adding due to the default change

### CODE REVIEW

**Comment by Maleclypse** [comment]
well that's madness.  the tests failed on the american 180 and hk_mp5 but neither of them are in your files at all

**Comment by Procyonae** [comment]
Ye this is kind of annoying bc peeps keep doing gun removals/renames that are moving stuff around

File: data/json/items/vehicle/manual.json [hunk 1]
```diff
@@ -12,16 +12,13 @@
     "volume": "500 ml",
     "category": "veh_parts",
     "price": "90 USD",
-    "price_postapoc": "1 USD",
-    "melee_damage": { "bash": 10 }
+    "price_postapoc": "1 USD"
```
Comment (Karol1223) [inline data/json/items/vehicle/manual.json:15]:
```suggestion
    "price_postapoc": "1 USD",
    "melee_damage": { "bash": 10 }
```

This has to come back or the tests will not let this go through. Why is this changed here anyway?

File: data/json/items/vehicle/manual.json [hunk 2]
```diff
@@ -12,16 +12,13 @@
...
   },
   {
     "type": "GENERIC",
     "id": "foot_crank",
     "copy-from": "hand_rims",
     "name": { "str": "foot crank" },
-    "description": "A pedal and gear assembly from a bicycle.",
-    "to_hit": -2,
-    "melee_damage": { "bash": 8, "cut": 2 }
+    "description": "A pedal and gear assembly from a bicycle."
```
Comment (Karol1223) [inline data/json/items/vehicle/manual.json:22]:
```suggestion
    "description": "A pedal and gear assembly from a bicycle.",
    "melee_damage": { "bash": 8, "cut": 2 }
```

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/items/vehicle/manual.json (hunk 1 above)
Comment (GuardianDll) [reply data/json/items/vehicle/manual.json:15]: because their dps test was already borked, i assumed hand rims is a bad weapon to be actually listed in dps test

Comment (Karol1223) [reply data/json/items/vehicle/manual.json:15]: > because their dps test was already borked, i assumed hand rims is a bad weapon to be actually listed in dps test

I don't disagree that they shouldn't be there, but the damage alone should probably not be scope creeped into this PR either...
